Add keyboard shortcuts support when using UIKit for Mac (Project Catalyst)

revolter opened the pull request:
Fixes #297.

NSExceptional commented:
Can you explain your changes and your approach? I'm wary of accepting something that uses the preprocessor to change the superclass of one of our classes

revolter commented:
Well, I know it's a bit hacky, but I tried to fix it with minimal changes. I forgot to add usage sample code though. You have to overwrite the `keyCommands` method in your `AppDelegate` and return `[[FLEXManager sharedManager] getKeyCommands]`.

The thing is that the only way to make it work is to return actual instances of `UIKeyCommand`.

NSExceptional commented:
Random question: what does FLEX even look like in a Catalyst app?

revolter commented:
Well, we don't develop the Mac app anymore, so I can't check it out, and I also don't remember.

But, at least in our case, since the Mac app was a window with a size (or at least an aspect ratio) of an iPad, both the app and FLEX looked like on a regular iPad.

NSExceptional commented:
Well since it's not a priority right now I'm gonna close this :P If you plan on deleting the associated branch, let me know so I can fork it; I don't want your work to go to waste if I do eventually get around to this

revolter commented:
> Well since it's not a priority right now I'm gonna close this :P

Sooo, I agreed with you, but then I thought that I could add Catalyst apps to the example project(s), so that contributors have the ability to test PRs like this one directly.

But after downloading the repo, I noticed that both the CocoaPods and the SPM example projects have a catalyst app added:

<img width="901" alt="image" src="https://user-images.githubusercontent.com/5748627/165587109-0e5c1c95-7811-4da3-bda7-10ec409cf37e.png">

Knowing this, I'll try to rebase this.

revolter committed:
Remove development team from SPM example project

revolter commented:
Actually, I think that I can run the app on my Mac because I have it's M1.

revolter committed:
Add keyboard shortcuts support when using UIKit for Mac (Project Catalyst)

revolter committed:
Fix SPM example project not using the local FLEX

https://forums.swift.org/t/how-to-add-local-swift-package-as-dependency/26457/8

revolter committed:
Enable Catalyst app for the example projects

revolter committed:
Enable keyboard shortcuts on Mac in the example app

https://developer.apple.com/documentation/apple-silicon/building-a-universal-macos-binary#Wrap-Platform-Specific-Code-with-Conditional-Compilation-Macros

revolter committed:
Add keyboard shortcuts support when running on ARM64

https://developer.apple.com/documentation/apple-silicon/building-a-universal-macos-binary#Wrap-Platform-Specific-Code-with-Conditional-Compilation-Macros

revolter committed:
Set spaces indentation for the example projects

revolter committed:
Simplify Mac shortcuts implementation

revolter committed:
Remove superfluous implementation

revolter commented:
Apparently, it works even if you don't override the `keyCommands` property getter, so the changes will be minimal. I don't know if anything changed since 2019, or if I was dumber and didn't notice this.

revolter commented:
@NSExceptional, this is now ready for review

NSExceptional commented:
Did you rebase it? Still says there are conflicts

revolter commented:
I had to rebase it yet again, which caused me to notice that you used `file:///Users/tanner/Repos/Reflex` in 8f995630, which will obviously not work for others, so I overwrite it with 5efec8c9.
